gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/FirestoreStorageManager.java

```diff
@@ -65,27 +68,35 @@ private StorageReference createReference(String path) {
     return createReference(remoteDestinationPath).getDownloadUrl();
   }
 
-  /** Upload file to Firebase Storage. */
   @Override
-  public Completable uploadMediaFromFile(File file, String remoteDestinationPath) {
-    return Completable.create(
+  public Flowable<Event<UploadProgress>> uploadMediaFromFile(
+      File file, String remoteDestinationPath) {
+    return Flowable.create(
         emitter ->
             createReference(remoteDestinationPath)
                 .putFile(Uri.fromFile(file))
                 .addOnCompleteListener(uploadTask -> emitter.onComplete())
-                .addOnFailureListener(emitter::onError)
+                .addOnPausedListener(
+                    taskSnapshot -> {
+                      emitter.onNext(Event.create(new UploadProgress(UploadState.PAUSED)));
```

Comment:
Instead of calling `new` directly, consider adding "static constructor" methods like `UploadProgress.paused()` and UploadProgress.inProgress(byteCount, bytesTransferred).